src/java.base/share/classes/java/lang/Class.java

```diff
@@ -642,7 +642,7 @@ private static Class<?> forName(Module module, String name, Class<?> caller) {
      * @since Valhalla
      */
     public boolean isIdentity() {
-        return (this.getModifiers() & Modifier.IDENTITY) != 0;
+        return isArray() || (this.getModifiers() & Modifier.IDENTITY) != 0;
```

Comment:
I think Class::getModifiers for an array class should have `IDENTITY` flag set.  I think it's a bug in the VM that should set `IDENTITY`.

```
jint ObjArrayKlass::compute_modifier_flags() const {
  // The modifier for an objectArray is the same as its element
  if (element_klass() == NULL) {
    assert(Universe::is_bootstrapping(), "partial objArray only at startup");
    return JVM_ACC_ABSTRACT | JVM_ACC_FINAL | JVM_ACC_PUBLIC;
  }
  // Return the flags of the bottom element type.
  jint element_flags = bottom_klass()->compute_modifier_flags();

  return (element_flags & (JVM_ACC_PUBLIC | JVM_ACC_PRIVATE | JVM_ACC_PROTECTED))
                        | (JVM_ACC_ABSTRACT | JVM_ACC_FINAL);
}
```

Reply:
The spec of Class.getModifiers() is explicit for arrays that only a few bits are defined.
The other bits are unspecified and may be implementation specific.
@rose00 thought it risky and too late to specify that now.

I'd suggest making the `Class.isIdentity()` and `Class.isValue()` native methods and have the VM implement the semantics.  Then it will not depend on the modifier bits. Mostly likely, these would become intrinsics.

Reply:
Alternatively, we could use `getClassAccessFlagsRaw` to return the VM specific flags to indicate that. 

I think checking `isArray` is a good interim solution.